src/ModelContextProtocol.Core/Client/McpClient.Methods.cs

```diff
@@ -570,13 +592,21 @@ async ValueTask<CallToolResult> SendRequestWithProgressAsync(
                     return default;
                 }).ConfigureAwait(false);
 
+            var metaWithProgress = meta is not null ? new JsonObject(meta) : new JsonObject();
+            metaWithProgress["progressToken"] = progressToken.Token switch
+                {
+                    string s => JsonValue.Create(s),
+                    long l => JsonValue.Create(l),
+                    _ => throw new InvalidOperationException("ProgressToken must be a string or long")
+                };
```

Comment:
We know it's backed by a string, as we created it on L581 from a string.
```suggestion
            metaWithProgress["progressToken"] = progressToken.ToString();
```